source/image-handler/image-request.ts

```diff
@@ -127,6 +128,10 @@ export class ImageRequest {
         result.lastModified = new Date(originalImage.LastModified).toUTCString();
       }
 
+      if (originalImage.ETag) {
+        result.eTag = originalImage.ETag;
+      }
```

Comment:
These lines add `ETag` of the original image, but when filters will be applied i.e. `ETag` of the original image is not equal to `ETag` of modified image. Also, there is code to check the `If-None-Match` header.